---
title: "loaded_pkg"
author: "hy"
date: "10/13/2016"
output: html_document
---

These packages need to be loaded in Iranman project

```{r, bioinformatic packages}
require(rtracklayer)
require(TxDb.Hsapiens.UCSC.hg19.knownGene)
require(BSgenome.Hsapiens.UCSC.hg19)
require(GenomicRanges)
require(GenomicFeatures)
require(genomation)
require(ChIPseeker)
```

=========================================================

```{r, plot packages}
require(ggplot2)
require(reshape2)
```

```{r}
##classical theme
hy.theme<-   theme(plot.title = element_text(color="black", size=20, face="bold.italic"),
                   axis.title.x = element_text( face="bold",size=14,colour = "black"),
                   axis.title.y = element_text(color="black", size=14, face="bold"),
                   legend.title =element_text(face = "bold", size = 14, color = "white"),
                   legend.text = element_text(face = "bold", size = 12),
                   axis.text.x = element_text(face = "bold",size=14),
                   axis.text.y = element_text(face="bold", size=14),
                   strip.text.x = element_text(face = "bold",size = 14),
                   strip.text.y = element_text(face = "bold",size = 14)
)
```



=========================================================
This Project self-define important funtion
```{r region.base.signal}
#The function can get a region signal coverage in single base
#necessary parameter: input regions, list of genome coverage
region.base.signal<- function(region,list.cov,strand=FALSE,weight.col=NULL,...){
  if(strand){
    cage.sg.p<-lapply(list.cov,function(x)ScoreMatrix(x,region[strand(region)=="+"],weight.col = weight.col))
    cage.sg.m<-lapply(list.cov,function(x)ScoreMatrix(x,region[strand(region)=="-"],weight.col = weight.col))
    cage.sg<- list(plus=cage.sg.p,minus=cage.sg.m)
    
  }else{
    cage.sg<-lapply(list.cov,function(x)ScoreMatrix(x,region,weight.col = weight.col))
    
  }
  return(cage.sg)
}


```


=========================================================
brilliant function or packages
```{r summarizeOverlaps}
###summarizeOverlaps like htseq-count

library(GenomicAlignments)
example(summarizeOverlaps)
features <- gr
reads <- reads
```

